### PR TITLE
Map implement size and as_array

### PR DESCRIPTION
**Pull request by FrankBlue**

### CODE REVIEW

File: src/object.rs [hunk 1]
```diff
@@ -278,3 +281,16 @@ impl Array {
     unsafe { v8__Array__Length(self) }
   }
 }
+
+impl Map {
+  pub fn size(&self) -> usize {
+    unsafe { v8__Map__Size(self) }
+  }
+  pub fn as_array<'sc>(
```
Comment (ry) [inline src/object.rs:289]: Please add the docstring from v8.h
```
/// Returns an array of length Size() * 2, where index N is the Nth key and
/// index N + 1 is the Nth value.
```

File: tests/test_api.rs [hunk 2]
```diff
@@ -2209,11 +2210,32 @@ fn value_checker() {
...
+        == v8::String::new(scope, "s").unwrap()
+    );
+    assert!(
+      map_array.get_index(scope, context, 2).unwrap()
+        == v8::String::new(scope, "v").unwrap()
+    );
+    assert!(
+      map_array.get_index(scope, context, 3).unwrap()
+        == v8::Number::new(scope, 8f64)
+    );
```
Comment (ry) [inline tests/test_api.rs:2238]: Please start a new test function for map rather than putting this in the proxy test.

Comment (FrankBlue) [reply tests/test_api.rs:2238]: Thanks, I add test code and comment

Comment (FrankBlue) [reply tests/test_api.rs]: I changed it

**Comment by ry** [comment]
Please run `cargo clippy --all-targets` and fix the errors.

**Pull request merged**